File: 0013-roman-to-integer/0013-roman-to-integer.cpp
class Solution {
public:
    int romanToInt(string s) {
        unordered_map<char,int> dict = {
            {'I',1},
            {'V',5},
            {'X',10},
            {'L',50},
            {'C',100},
            {'D',500},
            {'M',1000}
        };
        int res = 0,curr=0,prev=-1;
        for(int i=0;i<s.length();i++){
            curr=dict[s[i]];
            if(i>0 && curr>prev){
                curr-=prev;
                res-=prev;
            }
            res+=curr;
            prev=curr;
        }
        return res;
    }
};

File: 0013-roman-to-integer/0013-roman-to-integer.py
class Solution:
    def romanToInt(self, s: str) -> int:
        prev_val=None
        prev_char=None 
        roman_dict ={
            'I':1,'V':5,'X':10,
            'L':50,'C':100,'D':500,'M':1000
        }
        res = 0
        for i in s: 
            curr_val = roman_dict[i] 
            if prev_val and curr_val>prev_val: 
                res-=prev_val 
                res+=curr_val-prev_val
            else: 
                res+=curr_val 
            prev_val=curr_val
        return res
            